apps/authentication-service/src/shared/tokenUtils.ts

```diff
@@ -0,0 +1,69 @@
+import fs from "fs";
+import { decode, JwtPayload } from "jsonwebtoken";
+import os from "os";
+import path from "path";
+
+import { ConsoleColors } from "@/shared/enums";
+
+let token: string | undefined;
+const tokenFileName: string = "brkn-arrw.txt";
+
+export const tokenFile: string = path.join(os.tmpdir(), tokenFileName);
+
+export const getLocalToken = (): string | undefined => {
+  logToken(token);
+  return token;
+};
+
+export const setLocalToken = (newToken: string | undefined): void => {
+  token = newToken;
+};
+
+export const setCachedToken = (token: string): boolean => {
```

Comment:
Why did we separate setCachedToken and setLocalToken? Aren't they always used together in our non test code?

Reply:
As I started writing the tests, I realized thes cached token wasn't always being read and set as the local token. Sometimes cached was being ignored, even when it was valid, and a new one was being fetched.